tests/dat/actions/countdown.js

```diff
@@ -2,17 +2,19 @@
  * An action that invokes itself recursively, programmatically using the whisk
  * Javascript API.
  */
+var openwhisk = require('openwhisk')
+
 function main(params) {
+    var wsk = openwhisk({ignore_certs: true})
+
     var n = parseInt(params.n);
     console.log(n);
     if (n === 0) {
         console.log('Happy New Year!');
     } else if (n > 0) {
-        return whisk.invoke({
-            name : 'countdown',
-            parameters : {
-                n : n - 1
-            }
+        return wsk.actions.invoke({
+            actionName: process.env['__OW_ACTION_NAME'].split('/')[2],
```

Comment:
Noting here quick discussion with @jthomas: the js API expects name to be just the name and namespace to be the path. This is part of the confusion we created with abusing namespace to mean path. __OW_ACTION_NAME is the fully qualified name: (path, name). So we should either change the properties provided to the actions or change the API to accept a fully qualified name instead. Slightly favoring the latter since it's consistent with the CLI.